packages/@eventual/core/src/entity.ts

```diff
@@ -17,40 +18,32 @@ import {
 } from "./internal/service-spec.js";
 import type { ServiceContext } from "./service.js";
 
-export interface CompositeKey {
-  namespace: string;
-  key: string;
+export interface EntityQueryResultEntry<E extends EntityValue> {
+  entity: E;
+  version: number;
 }
 
-export interface EntityListResult<Entity> {
-  entries?: { key: string; entity: Entity; version: number }[];
+export interface EntityQueryResult<E extends EntityValue> {
+  entries?: EntityQueryResultEntry<E>[];
   /**
    * Returned when there are more values than the limit allowed to return.
    */
   nextToken?: string;
 }
 
-export interface EntityListKeysResult {
-  /**
-   * Keys that match the provided prefix. If using composite keys, this will only be the key part (not the namespace).
-   */
-  keys?: string[];
-  /**
-   * Returned when there are more values than the limit allowed to return.
-   */
-  nextToken?: string;
-}
-
-export interface EntityListRequest {
+export interface EntityQueryRequest<
+  E extends EntityValue,
+  Partition extends EntityKeyField<E>
+> {
   /**
-   * Namespace to retrieve values for.
+   * Partition key to retrieve values for.
    *
    * @default - retrieve values with no namespace.
    */
-  namespace?: string;
+  partition: E[Partition];
```

Comment:
This seems like it only accepts a single value?

Reply:
Have not pushed a commit since the last time we talked. Should be ready soon.